### PR TITLE
FORMS-200 # Date model value doesn't propagate to pickadate modal picker

### PR DESCRIPTION
**Pull request by GeetaSajwan**
- element.val() will be reflected in picker
- element.val('2015-10-02') should use ISO 8601 format
- input.val() will be using date element format or default format i.e ISO 8601
- error will be thrown in case invalid format is used

### CODE REVIEW

File: test/1_date_time/test.js [hunk 1]
```diff
@@ -264,11 +281,72 @@ define(['BlinkForms', 'testUtils'], function (Forms, testUtils) {
...
+        'dateTimeHiddenNowP',
+        'dateTimeHddenNowPM',
+        'dateTimePickerNone',
+        'dateTimePickerNow',
+        'dateTimePickerNowP',
+        'dateTimePckerNowPM'
+      ];
+
+      test('element value assignment reflected in picker, model, view ', function (done) {
+        setTimeout(function () {
```
Comment (simonmarklar) [inline test/1_date_time/test.js:307]: is the timeout necessary?

File: forms/jqm/elements/date.js [hunk 2]
```diff
@@ -83,7 +84,19 @@ define(function (require) {
...
+      var value = $(event.target).val();
+
+      try {
+        this.model.isInvalidFormat(dateFormat, value);
+      } catch (err) {
+        window.console.log(err);
+        return;
+      }
+      if (value) {
+        value = moment(value, dateFormat).format('YYYY-MM-DD');
```
Comment (simonmarklar) [inline forms/jqm/elements/date.js]: it might be worth making 'YYYY-MM-DD' and 'yyyy-mm-dd' into constants

**Comment by simonmarklar** [comment]
im not really fussed about my comments being done, they are more suggestions than anything :)

working with dates is painful. The code LGTM :+1:

File: test/1_date_time/test.js (hunk 1 above)
Comment (GeetaSajwan) [inline test/1_date_time/test.js:307]: Tests were failing sometime so I used it.

File: forms/jqm/elements/date.js (hunk 2 above)
Comment (GeetaSajwan) [inline forms/jqm/elements/date.js]: Yes, you are right I make them constants.

File: test/1_date_time/test.js (hunk 1 above)
Comment (simonmarklar) [inline test/1_date_time/test.js:307]: ok no worries :)

i've noticed that sometimes things fail due to timing. its pretty painful when it seems random :(

**Comment by GeetaSajwan** [comment]
Yes that's the reason i added setTimeout. Thanks

File: forms/jqm/elements/date.js (hunk 2 above)
Comment (simonmarklar) [inline forms/jqm/elements/date.js]: sorry to be so nit picky but you should move the `var defaultFormat` and `var pickerFormat` declarations out of the functions and above the line

```
var DateElementView = ElementView.extend({
```

so they are only declared once, not every time the functions are run

and ron will probably ask that the names be uppercased like  `DEFAULT_FORMAT`

**Comment by jokeyrhyme** [comment]
The unit tests seem to be failing on Travis CI. Does `npm test` pass on your local machine?

**Comment by simonmarklar** [comment]
:+1: LGTM

**Pull request merged**